src/CircleEIP1193Provider.ts

```diff
@@ -1,7 +1,19 @@
 import axios, { AxiosInstance } from 'axios';
+import { encryptEntitySecret } from "./utils/helpers"
 
 const API_URL = 'https://api.circle.com/v1/w3s';
 
+interface WalletSetResponse {
+  data: {
+    walletSet: {
+      id: string;
+      custodyType: string;
+      updateDate: Date;
+      createDate: Date;
+    };
+  };
+}
+
```

Comment:
Nice, but this is unused. You can add it to ` this.client.post` in your method below. Reference how `PublicKeyResponse` is used in its method.